tap_facebook/streams/ad_insights.py

```diff
@@ -67,6 +67,15 @@ def __init__(self, *args, **kwargs) -> None:  # noqa: ANN002, ANN003
     def primary_keys(self) -> list[str] | None:
         return ["date_start", "account_id", "ad_id"] + self._report_definition["breakdowns"]
 
+    @primary_keys.setter
+    def primary_keys(self, new_value: list[str] | None) -> None:
+        """Set primary key(s) for the stream.
+
+        Args:
+            new_value: TODO
+        """
+        self._primary_keys = new_value
```

Comment:
@pnadolny13 TIL the identity of the property object changes if you override it. Are you seeing something like `AttributeError: property 'x' of 'Class' object has no setter`?

Reply:
@edgarrmondragon I'm seeing `AttributeError: can't set attribute 'primary_keys'. Did you mean: '_streams'?` when its called [here](https://github.com/meltano/sdk/blob/b367eb5e7b481c5498f117bf3522f6e5d86e6584/singer_sdk/streams/core.py#L1231)

Reply:
If that's the case, this is something to call out in the docs.

PS: I would've expect mypy to catch this, so we might wanna make sure to run it in CI.